e2e/support/commands/downloads/downloadUtils.js

```diff
@@ -0,0 +1,114 @@
+const fs = require("fs");
+
+const path = require("path");
+
+const removeDirectory = path => {
+  try {
+    fs.rmdirSync(path, { maxRetries: 10, recursive: true });
+  } catch (error) {
+    console.log("Error while removing directory", path, error);
+  }
+  return null;
+};
+
+const deleteDownloadsFolder = () =>
+  cy.task("removeDirectory", Cypress.config("downloadsFolder"));
+
+const isFileExist = path => fs.existsSync(path);
+
+const findFiles = ({ path, fileName }) => {
+  if (!fs.existsSync(path)) {
+    return null;
+  }
+
+  return fs
+    .readdirSync(path)
+    .filter(file => file.includes(fileName) && isDownloaded(file));
+};
+
+const isDownloaded = file => !file.endsWith(".crdownload");
+
+// slightly modified version of cy-verify-download
+// https://github.com/elaichenkov/cy-verify-downloads/blob/master/src/index.js
+const verifyDownload = (fileName, options) => {
```

Comment:
we were using a library for this functionality, but it had some bugs for more complex cases, so I just copied the source here and changed what I wanted instead of forking it.